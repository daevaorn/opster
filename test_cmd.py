#!/usr/bin/env python

import finaloption

config_opts=[('c', 'config', 'webshops.ini', 'config file to use')]


@finaloption.command(config_opts)
def initdb(config):
    """Initialize database"""
    pass


@finaloption.command(options=config_opts + [
    ('h', 'host', 'localhost', 'The host for the application.'),
    ('p', 'port', 5000, 'The port for the server.'),
    ('', 'nolint', False, 'Do not use LintMiddleware')
])
def runserver(**opts):
    """Run development server"""
    print opts


finaloption.dispatch()
