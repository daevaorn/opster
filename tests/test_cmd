#!/bin/sh

python test_cmd.py
python test_cmd.py help runserver
python test_cmd.py initdb
