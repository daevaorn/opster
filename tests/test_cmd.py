#!/usr/bin/env python

import opster

config_opts=[('c', 'config', 'webshops.ini', 'config file to use')]

@opster.command(config_opts)
def initdb(config):
    """Initialize database"""
    pass

@opster.command(config_opts)
def runserver(listen=('l', 'localhost', 'ip to listen on'),
              port=('p', 5000, 'port to listen on'),
              **opts):
    """Run development server"""
    print locals()

if __name__ == '__main__':
    opster.dispatch()
