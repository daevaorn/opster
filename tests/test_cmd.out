usage: test_cmd.py <command> [options]

commands:

 help       Show help for a given help topic or a help overview
 initdb     Initialize database
 runserver  Run development server
test_cmd.py runserver [OPTIONS] 

Run development server

options:

 -l --listen  ip to listen on (default: localhost)
 -p --port    port to listen on (default: 5000)
 -c --config  config file to use (default: webshops.ini)
 -h --help    display help
